### PR TITLE
Create the base class to authenticate and submit entities with Json API

### PR DESCRIPTION
**Pull request by pfrilling**
## Summary
This introduces the `EcmsApiBase` class that will handle:
- Getting access tokens from the endpoint sites
- Preparing and submitting entities to the endpoint sites

This is an abstract class by default as two new services will extend this class to get the full functionality needed. 

## Metadata
| Question | Answer |
|----------|--------|
| Did you use a meaningful pull request title? | yes
| Did you apply meaningful labels to the pull request? | yes
| Did you perform a self review first? | yes
| Documentation reflects changes? | yes
| `CHANGELOG` reflects changes? | yes
| Unit/Functional tests reflect changes? | yes
| Did you perform browser testing? | yes
| Risk level | Medium
| Relevant links | [RIG-61](https://thinkoomph.jira.com/browse/rig-61)

### CODE REVIEW

File: ecms_base/modules/custom/ecms_api/src/EcmsApiBase.php [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+   *   The jsonapi_extras.entity.to_jsonapi service.
+   */
+  public function __construct(ClientInterface $httpClient, EntityToJsonApi $entityToJsonApi) {
+    $this->httpClient = $httpClient;
+    $this->entityToJsonApi = $entityToJsonApi;
+  }
+
+  /**
+   * Get an access token to authorize the request.
+   *
```
Comment (ben-hamelin) [inline ecms_base/modules/custom/ecms_api/src/EcmsApiBase.php:84]: Argument comments should be added here

**Comment by ben-hamelin** [comment]
@pfrilling Just the one minor note on the function arguments comments, otherwise looks good!

File: ecms_base/modules/custom/ecms_api/src/EcmsApiBase.php (hunk 1 above)
Comment (pfrilling) [reply ecms_base/modules/custom/ecms_api/src/EcmsApiBase.php:84]: Good catch. Not sure how phpcs missed that?

**Comment by ben-hamelin** [comment]
Ok looks good!
/approved

**Pull request merged**